feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/radio.c

```diff
@@ -330,6 +341,8 @@ static bool radio_state_update(struct uci_section *s, struct schema_Wifi_Radio_C
 			SCHEMA_SET_INT(rstate.channel, chan);
 		else
 			SCHEMA_SET_INT(rstate.channel, blobmsg_get_u32(tb[WDEV_ATTR_CHANNEL]));
+			radio_fixup_set_primary_chan(rstate.if_name,
```

Comment:
Indentation is off or missing {} on else cause.

Why update this on the radio_state_update() instead of on a config operation?

Reply:
We update it in the config too, but in case there is no config change yet like when bootup, the current config is got from the uci using tb[WDEV_ATTR_CHANNEL]. So our fixup variable and uci wireless config are same always. The new channel coming from the vif config  is compared with this before setting it.